### PR TITLE
Convert System.Net.Http.Native to C

### PR DESCRIPTION
**Pull request by kasper3**
<del>`git mv` (to rename a file) has a limit, if the change exceeds to some thousand characters, it doesn't show it as inline diff, but instead as if a file was deleted and a new one was added. This is what happened in `pal_easy.c`.</del>

### CODE REVIEW

**Comment by stephentoub** [comment]
Can you please do it then as separate commits, one that renames and one that edits, so that we can more easily review?

**Comment by kasper3** [comment]
@stephentoub, i have split the commits.

**Comment by kasper3** [comment]
test OSX x64 Debug Build please

File: src/Native/Unix/System.Net.Http.Native/pal_easy.c [hunk 1]
```diff
@@ -0,0 +1,360 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (kasper3) [inline src/Native/Unix/System.Net.Http.Native/pal_easy.c]: btw, what is the rational of calling this file `pal_easy`? 360 lines of code in this file isn't that much easier when compared to other five liners c files. 😄

Comment (kasper3) [reply src/Native/Unix/System.Net.Http.Native/pal_easy.c]: btw, what is the rationale behind calling this file `pal_easy`? 360 lines of code in this file isn't that much easier when compared to other five liners c files. 😄

Comment (kasper3) [inline src/Native/Unix/System.Net.Http.Native/pal_easy.c]: btw, what is the rationale behind calling this file `pal_easy`? 360 lines of code in this file isn't that much easier when compared to other five liners c files. 😄

Comment (stephentoub) [reply src/Native/Unix/System.Net.Http.Native/pal_easy.c]: "easy" is what libcurl calls these APIs.

File: src/Native/Unix/System.Net.Http.Native/pal_curlinit.h [hunk 2]
```diff
@@ -15,4 +16,4 @@
  *
  * Returns 0 on success and non-zero on failure.
  */
-extern "C" int32_t HttpNative_EnsureCurlIsInitialized();
+DLLEXPORT int32_t HttpNative_EnsureCurlIsInitialized(void);
```
[the remark this answers is not in the document]
Comment (kasper3) [reply src/Native/Unix/System.Net.Http.Native/pal_curlinit.h]: Would it be ok to skip it for now as two other PRs are merged without C externs? I intend to send PRs for remaining two directories straight up.

**Comment by kasper3** [comment]
@jkotas, can you please take a look?

File: src/Native/Unix/System.Net.Http.Native/pal_curlinit.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkotas) [reply src/Native/Unix/System.Net.Http.Native/pal_curlinit.h]: FWIW, we are not consistent already - we are missing them in several files where they are not needed, e.g. https://github.com/dotnet/corefx/blob/master/src/Native/Unix/System.Net.Security.Native/pal_gssapi.h

**Comment by jkotas** [comment]
Do we have coverage for the Curl handler in the CI?

File: src/Native/Unix/System.Net.Http.Native/pal_curlinit.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/Native/Unix/System.Net.Http.Native/pal_curlinit.h]: > we are not consistent already

😦

**Comment by stephentoub** [comment]
> Do we have coverage for the Curl handler in the CI?

Yes, but it's best to run outer loop, too.

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Linux x64 Release Build please

**Comment by kasper3** [comment]
Test failure is https://github.com/dotnet/corefx/issues/30230, 

https://mc.dot.net/#/user/kasper3/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/509af4b53826277ba34322022679af3daee060b4/workItem/System.Drawing.Common.Tests/analysis/xunit/System.Drawing.Tests.Graphics_DrawLineTests~2FDrawLines_PointFs

**Pull request merged**